app/controllers/loans.js

```diff
@@ -0,0 +1,350 @@
+/**
+  Loans Controller 
+*/
+
+//3rd party modules
+var winston = require('winston');
+var express = require('express');
+var mongoose = require('mongoose');
+
+//own modules
+var DefaultController = require('./');
+
+var Item = undefined;
+var Loan = undefined;
+var defaultCtrl = undefined;
+
+var Controller = function(){
+  Item = mongoose.model('Item');
+  Loan = mongoose.model('Loan');
+  defaultCtrl = new DefaultController(Loan, 'Loan');
+
+  // Define route params
+  this.paramFormat = defaultCtrl.format();
+  this.paramLoan = defaultCtrl.modelParam();
+  
+  // Define handlers for rest calls
+  this.get = defaultCtrl.get;
+  this.find = defaultCtrl.find;
+  this.create = validateAndCreate;
+  
+  this.update = validateAndUpdate;
+  this.remove = defaultCtrl.remove;
+
+  this.me = function(req, res) {
+    Loan.find({loaner: req.user})
+    .populate('items.item')
+    .exec(function(err, loans) {
+      res.json(loans);
+    });
+  }
+
+  return this;
+}
+
+// Iterators
+function makeArrayIterator(array) {
```

Comment:
this looks very weird to me.. maybe json schema validator could do same thing with less of code..?